template/macros.jinja

```diff
@@ -43,22 +42,30 @@
             - salt://xxx/files/default/etc/yyy/zzz.conf.jinja
           - template: jinja
   #}
-  {%- set path_prefix = prefix|replace(':', '/') %}
-  {%- set files_switch_list = salt['pillar.get'](prefix ~ ':files_switch',
-                                           default_files_switch) %}
+  {%- set formula = 'template' %}
+  {%- set path_prefix = salt['config.get'](formula ~ ':tofs:path_prefix', formula) %}
+  {%- set files_switch_list = salt['config.get'](
+      formula ~ ':files_switch',
+      default_files_switch
+  ) %}
   {%- for grain in files_switch_list if grains.get(grain) is defined %}
     {%- for file in files %}
-      {%- set url = '- salt://' ~ '/'.join([path_prefix,
-                                            'files',
-                                            grains.get(grain),
-                                            file.lstrip('/')]) %}
+      {%- set url = '- salt://' ~ '/'.join([
+          path_prefix,
+          salt['config.get'](formula ~ ':tofs:dirs:files', 'files'),
+          grains.get(grain),
+          file.lstrip('/')
+      ]) %}
 {{ url | indent(indent_width, true) }}
     {%- endfor %}
   {%- endfor %}
-    {%- for file in files %}
-      {%- set url = '- salt://' ~ '/'.join([path_prefix,
-                                            'files/default',
-                                            file.lstrip('/')]) %}
+  {%- for file in files %}
+    {%- set url = '- salt://' ~ '/'.join([
+        path_prefix,
+        salt['config.get'](formula ~ ':tofs:dirs:files', 'files'),
+        salt['config.get'](formula ~ ':tofs:dirs:default', 'default'),
```

Comment:
Using `config.get` to make the `'default'` directory configurable or use `'default'` by default.

Reply:
I find the `'- salt://'` weird. I would have done `salt://` and adding the dash in the loop, to have the 'url' really be an url, not an item in a YAML list

Reply:
maybe the same for `salt://' ~ '/'`, why not use `salt:///` directly?

Reply:
That particular section was just direct refactoring, didn't look at it in any detail.

So looking now, this is being used to populate the list under the `source` of `file.managed`, thus requiring the dash `-` prefixed.